core/app/c/[communitySlug]/forms/[formSlug]/edit/EditFormTitleButton.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+
+import React, { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Button } from "ui/button";
+import { Dialog, DialogContent, DialogFooter, DialogTitle, DialogTrigger } from "ui/dialog";
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from "ui/form";
+import { Check, X } from "ui/icon";
+import { Input } from "ui/input";
+import { toast } from "ui/use-toast";
+
+import { didSucceed, useServerAction } from "~/lib/serverActions";
+import { updateForm } from "./actions";
+
+const editFormTitleSchema = z.object({
+	name: z.string().min(1, "Form name is required"),
+});
+
+type NotificationProps = {
+	message: string;
+	success?: boolean;
+};
+
+const NotificationComponent: React.FC<NotificationProps> = ({ message, success = true }) => {
+	return (
+		<div
+			className={`mx-auto flex items-center ${
+				success
+					? "border-green-200 bg-green-100 text-green-800"
+					: "border-red-200 bg-red-100 text-red-800"
+			}`}
+		>
+			<div className="mr-3 text-2xl">{success ? <Check /> : <X />}</div>
+			<div className="text-lg font-semibold">{message}</div>
+		</div>
+	);
+};
+
+const EditFormTitleButton = ({ formId, name }: { formId: string; name: string }) => {
+	const [isOpen, setIsOpen] = useState(false);
+
+	const form = useForm<z.infer<typeof editFormTitleSchema>>({
+		resolver: zodResolver(editFormTitleSchema),
+		defaultValues: {
+			name,
+		},
+	});
+
+	const runUpdateFormTitle = useServerAction(updateForm);
+	const onSubmit = async (data: z.infer<typeof editFormTitleSchema>) => {
+		const result = await runUpdateFormTitle({ formId, name: data.name });
+		if (didSucceed(result)) {
+			toast({
+				className: "bg-green-100 border-green-200 text-green-800",
+				description: <NotificationComponent message="Name successfully updated" />,
+			});
+			setIsOpen(false);
+		}
+	};
+	return (
+		<Dialog onOpenChange={setIsOpen} defaultOpen={false} open={isOpen} modal={true}>
+			<DialogTrigger asChild>
+				<Button
+					variant="link"
+					className="ml-2 text-sm text-blue-500 hover:text-blue-600 hover:underline"
+				>
+					Edit
+				</Button>
+			</DialogTrigger>
+			<DialogContent>
+				<DialogTitle>Edit Name</DialogTitle>
+				<hr />
+				<Form {...form}>
+					<form onSubmit={form.handleSubmit(onSubmit)}>
+						<FormField
+							control={form.control}
+							name="name"
+							render={({ field }) => (
+								<FormItem>
+									<FormLabel>Form Name</FormLabel>
+									<FormControl>
+										<Input placeholder="Name" {...field} />
+									</FormControl>
+									<FormMessage />
+								</FormItem>
+							)}
+						/>
+						<DialogFooter className="flex justify-end">
```

Comment:
There needs to be some margin between the footer and the form above 
![image](https://github.com/user-attachments/assets/b0fa98f6-6e9a-4967-834a-715b018596aa).

The cancel button should also be a different variant, and it looks like the gap between the buttons should be smaller

![image](https://github.com/user-attachments/assets/9463ba91-c5e4-477d-8892-72e6075c0c36)